tools/site_cobble/rdl_pkg/json_dump.py

```diff
@@ -108,6 +118,10 @@ def convert_addrmap_or_regfile(rdlc: RDLCompiler, obj: Union[AddrmapNode, Regfil
             json_child = convert_addrmap_or_regfile(rdlc, child)
         elif isinstance(child, RegNode):
             json_child = convert_reg(rdlc, child)
+        elif isinstance(child, MemNode):
+            json_child = convert_mem(child)
+        else:
+            raise RuntimeError("Unknown child type seen during JSON generation.")
```

Comment:
Without enumerating the full set of child types, the previous value of `json_child` was carried over from the previous loop iteration if it was not overwritten. This meant that before the `MemNode` branch existed, it would just repeat the previous `RegNode` for every instance of `MemNode`, resulting in #37